content/content/Examples/stateless-app-mirror.md

```diff
@@ -0,0 +1,290 @@
+in a non-destructive way. Going forward you'll be working against these
+manifests on disk without impacting the active resources in the "source"
+cluster.
+
+# Generate Transformations
+
+Crane's `transform` command generates tranformations in the form of JSON patches
+and stores those patches on disk in `transform` directory (unless overridden via
+`--transform-dir`).
+
+```bash
+crane transform
+```
+
+And verify that it behaved as we expect by checking out the `transform`
+directory:
+
+```bash
+$ tree -a transform
+transform
+└── resources
+    └── guestbook
+        ├── transform-ConfigMap_guestbook_kube-root-ca.crt.yaml
+        ├── transform-Deployment_guestbook_frontend.yaml
+        ├── transform-Deployment_guestbook_redis-master.yaml
+        ├── transform-Deployment_guestbook_redis-slave.yaml
+        ├── transform-Secret_guestbook_default-token-5vsrb.yaml
+        ├── transform-ServiceAccount_guestbook_default.yaml
+        ├── transform-Service_guestbook_frontend.yaml
+        ├── transform-Service_guestbook_redis-master.yaml
+        ├── transform-Service_guestbook_redis-slave.yaml
+        ├── .wh.Endpoints_guestbook_frontend.yaml
+        ├── .wh.Endpoints_guestbook_redis-master.yaml
+        ├── .wh.Endpoints_guestbook_redis-slave.yaml
+        ├── .wh.EndpointSlice_guestbook_frontend-bkqbs.yaml
+        ├── .wh.EndpointSlice_guestbook_redis-master-hxr5k.yaml
+        ├── .wh.EndpointSlice_guestbook_redis-slave-8wt7z.yaml
+        ├── .wh.Pod_guestbook_frontend-5fd859dcf6-5nvbm.yaml
+        ├── .wh.Pod_guestbook_frontend-5fd859dcf6-j8w94.yaml
+        ├── .wh.Pod_guestbook_frontend-5fd859dcf6-s9x8p.yaml
+        ├── .wh.Pod_guestbook_redis-master-55d9747c6c-6f9bz.yaml
+        ├── .wh.Pod_guestbook_redis-slave-5c6b4c5b47-jnrsr.yaml
+        ├── .wh.Pod_guestbook_redis-slave-5c6b4c5b47-xz776.yaml
+        ├── .wh.ReplicaSet_guestbook_frontend-5fd859dcf6.yaml
+        ├── .wh.ReplicaSet_guestbook_redis-master-55d9747c6c.yaml
+        └── .wh.ReplicaSet_guestbook_redis-slave-5c6b4c5b47.yaml
+
+2 directories, 24 files
+```
+
+**NOTE**
+
+What `crane transform` is doing, is iterating through the configured plugins and
+running them against the exported resources from the prevoius step. You can see
+which plugins are configured with `crane transform list-plugins` **and** you can
+see optional arguments to those plugins with `crane transform optionals`.
+
+Now is a great time to explore what plugins you can configure with `crane
+plugin-manager list`, install one, and maybe customize the exported resources
+with something like:
+
+```bash
+crane transform --optional-flags="add-annotations=custom-crane-annotation=foo"
+```
+
+If the flags start to get tough to manage via the command-line, you can also
+specify a `--flags-file` that might look something like:
+
+```yaml
+debug: false
+export-dir: myExport
+transform-dir: myTransform
+output-dir: myOutput
+optional-flags:
+  add-annotations: "custom-crane-annotation=foo"
```

Comment:
Actually, this should be specified as a map:
```
  add-annotations:
    "custom-crane-annotation": "foo"
```

Reply:
Map and slice params have to be encoded as comma-delimited strings when using a CLI flag, but in yaml they're specified as normal slices or maps.

Reply:
Yeah, actually the quotes aren't required here. I was getting confused between yaml and json here. I think this is good.